### ISSUE TITLE
Non interactive installation for Extension asks for name

### ISSUE THREAD

**Issue by gugahoi** [firebase/firebase-tools#3885]
### [REQUIRED] Environment info

**firebase-tools:** 9.21.0

**Platform:** MacOS

### [REQUIRED] Test case

When installing an extension and using `--non-interactive` and `--force`, the CLI still asks for an instance name. Can this be provided as a flag or param?

### [REQUIRED] Steps to reproduce

```
───────┬─────────────────────────────────────────────────────────────────────────────────
       │ File: extensions/info-to-big-query.env
───────┼─────────────────────────────────────────────────────────────────────────────────
   1   │ LOCATION=australia-southeast1
   2   │ DATASET_LOCATION=australia-southeast1
   3   │ COLLECTION_PATH=data/{userId}/info
   4   │ DATASET_ID=users_data
   5   │ TABLE_ID=info
   6   │ TABLE_PARTITIONING=MONTH
───────┴─────────────────────────────────────────────────────────────────────────────────
```

```sh
firebase ext:install firebase/firestore-bigquery-export --params=./extensions/info-to-big-query.env --force --non-interactive
```

Eventually leads to this:

```
i  extensions: Stream Collections to BigQuery will enable the following APIs for project xxxxxxxxxx
-bigquery-json.googleapis.com: Mirrors data from your Cloud Firestore collection in BigQuery.
i  extensions: Stream Collections to BigQuery will be granted the following access to project xxxxxxxx
-BigQuery Data Editor (Access to edit all the contents of datasets)
? Please enter a new name for this instance: (firestore-bigquery-export-8tw2) 
```

Which is interactive and errors the command.

### [REQUIRED] Expected behavior

The command runs successfully without needing user interaction

### [REQUIRED] Actual behavior

The command stops and asks for an instance name. I have been unable to find documentation on how to supply this via the CLI or params file so it seems impossible to have a non interactive installation of this extension.

**Comment by mothman11** [comment]
I agree, this would be a great addition to non-interactive.

In addition to this issue, I have noticed that when we have a project that has an extension name without the 4 character code at the end i.e. "firestore-bigquery-export", running an ext:install with --force --non-interactive will always update that extension, rather than giving the prompt to enter a new name as above.

![image](https://user-images.githubusercontent.com/54343728/142755973-2fd1ae64-1971-4ae5-a34c-8ffba9c5db12.png)

![image](https://user-images.githubusercontent.com/54343728/142755959-4bd66c7b-5082-4542-8ef5-bc090bfa80f9.png)
(no prompt to enter new extension instance name)

This has let to cases where we have overwritten an extension instance that's pointing to a different collection.

**Comment by krishnadubeyQ** [comment]
Any Update on this one?

**Comment by joelsdc** [comment]
As a workaround... I'm doing this:

```sh
echo "the-extension-name-I-want" | firebase ext:install ....
```

This method allows to set the name, and doesn't require human input.

**Comment by joehan** [comment]
This is fixed now - as of firebase-tools@11.0.0, we've changed how extensions deploys work to be better suited to noninteractive environments. Now, `firebase ext:install` sets up your extensions manifest, and `firebase deploy` actually deploys it onto your project. You should run `ext:install` interactively while you are building your CI/CD pipeline, and then run `firebase deploy` from the noninteractive environment. https://firebase.google.com/docs/extensions/manifest  has more details on how to do this.

I'm gonna close this now - if you are still on older versions of firebase-tools, @joelsdc workaround is a reasonable solution.

**Comment by gugahoi** [comment]
Thanks for the updates. It seems much easier to manage extensions now!
